test/Test/Iris/Interactive/Question.hs

```diff
@@ -0,0 +1,36 @@
+module Test.Iris.Interactive.Question (questionSpec) where
+
+import Test.Hspec (Spec, describe, it, shouldReturn)
+
+import Control.Monad (forM_)
+import Data.Text (unpack)
+
+import Iris.Interactive.Question (
+    -- under test
+    YesNo (..),
+    parseYesNo,
+ )
+
+questionSpec :: Spec
+questionSpec =
+    describe "Question - parse YesNo" $ do
+        describe "should parse to Just Yes" $ do
+            forM_ ["y", "Y", "YES", "yes", "Yes"] $
```

Comment:
Let's test all possible combinations of upper- and lowercase letters.

I can think of the following way to generate a list of all cases we want to handle. Something like this:

```haskell
yesAnswers :: [Text]
yesAnswers = "y" : "Y" : [y <> e <> s | y <- ["y", "Y"], e <- ["e", "E", ""], s <- ["s", "S"]]

Reply:
rewriting the test so all the cases are reported on one row 
```
    Question - parse YesNo
      elementes should parse to Just Yes
        ["y","Y","yes","yeS","yEs","yES","ys","yS","Yes","YeS","YEs","YES","Ys","YS"] [✔]
      elementes should parse to Just No
        ["n","N","NO","no","No","nO"] [✔]
      elementes should parse to Nothing
        ["a","ye","NOone","yesterday","oui"] [✔]
```

a failure becomes:

```
Failures:

  test/Test/Iris/Interactive/Question.hs:34:54:
  1) Iris.Interactive, Question - parse YesNo, elementes should parse to Just No, ["ne","N","NO","no","No","nO"]
       Actual list is not a permutation of expected list!
         expected list contains:   [("ne",Just No), ("N",Just No), ("NO",Just No), ("no",Just No), ("No",Just No), ("nO",Just No)]
         actual list contains:     [("ne",Nothing), ("N",Just No), ("NO",Just No), ("no",Just No), ("No",Just No), ("nO",Just No)]
         the missing elements are: [("ne",Just No)]
         the extra elements are:   [("ne",Nothing)]
```
so it's still easy to see what went wrong in detail.

like?